app/src/main/java/com/alexbar/layoutbasic/movies_app/screen/trending/TrendingMediaScreen.kt

```diff
@@ -1,49 +1,65 @@
 package com.alexbar.layoutbasic.movies_app.screen.trending
 
-import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.tooling.preview.Preview
 import com.alexbar.layoutbasic.movies_app.model.Media
-import com.alexbar.layoutbasic.movies_app.screen.detail.MediaDetailScreen
+import com.alexbar.layoutbasic.movies_app.networking.RetrofitInstance.mediaService
 import com.alexbar.layoutbasic.movies_app.screen.trending.widgets.DisplayMedia
 import com.alexbar.layoutbasic.movies_app.screen.trending.widgets.ShowAllMediaGrid
 
 @Composable
 fun TrendingMediaScreen(
-    movieList: List<Media>,
-    seriesList: List<Media>,
     favorites: List<Media>,
+    onMediaItemSelected: (Media) -> Unit,
     onFavoriteMediaSelected: (Media) -> Unit
 ) {
+    var movieList by remember { mutableStateOf<List<Media>>(emptyList()) }
+    var seriesList by remember { mutableStateOf<List<Media>>(emptyList()) }
+
+    LaunchedEffect(Unit) {
+        val responseMovies = mediaService.getTrendingMovies()
+        val responseSeries = mediaService.getTrendingSeries()
+
+        if (responseMovies.isSuccessful) {
+            movieList = responseMovies.body()?.results ?: emptyList()
+        } else {
+            println("Error")
+        }
+
+        if (responseSeries.isSuccessful) {
+            seriesList = responseSeries.body()?.results ?: emptyList()
+        } else {
+            println("Error")
+        }
+    }
+
```

Comment:
@ezatpanah  In both the **TrendingMediaScreen** and **MixMediaScreen**, I use the same lists obtained from the API. My question is whether it's okey for each screen to have this logic, even if it's duplicated, or if I should centralize it in the NavGraph or another location and pass the data to them through the constructor.